etc/concourse/scripts/common-functions

```diff
@@ -48,23 +48,27 @@ function restartAppsWithRetry {
 
 function waitForServices {
   echo "Waiting for DB creation or update ..."
-  until [ "$(cf services | grep '\(create\|update\) succeeded\|\(create\|update\) failed' | wc -l)" == "$1" ]; do
+  services=$1
+  for service in "${services[@]}"
+  do
+    while [[ -z $(cf service $service | grep '\(create\|update\) succeeded\|\(create\|update\) failed') ]]; do
     sleep 3s
   done
-  if [ "$(cf services | grep '\(create\|update\) succeeded' | wc -l)" == "$1" ]; then
-    echo "DB creation or update finished successfully."
+  if [[ -n $(cf service $service | grep '\(create\|update\) succeeded') ]]; then
```

Comment:
We need to indent the if so it is more visible the checks are executed in the for. Not sure if moving the do (line 53) next to the for cycle would help to achieve this.